### PR TITLE
feat-andrew --> master

### PR DESCRIPTION
**Pull request by andrewwinkler**
Addition and implementation of more CLIs.
Renaming of class "ImportTV" to "AllTV".
Rename variable "episodes" in class AllTV to "shows".

### CODE REVIEW

File: src/com/andrew/plex/cli/Cli.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			return false;
+			
+		} else {
+			
+			System.out.println("Not a show!");
+			return true;
+		}
+	}
+	
+	private String lengthHumanReadable( int l) {
```
Comment (thomashw) [inline src/com/andrew/plex/cli/Cli.java]: Fix spacing.

File: src/com/andrew/plex/cli/Cli.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package com.andrew.plex.cli;
```
Comment (thomashw) [inline src/com/andrew/plex/cli/Cli.java]: More comments... everywhere! Really important.

Javadoc is a pretty popular way to document classes/methods. A javadoc parser can automatically parse your source code and turns it into external documentation. All this documentation was generated from parsing documentation within source files: https://www.javadoc.io/doc/org.springframework/spring-core/5.0.2.RELEASE

It's a really handy way for other developers to understand how to use your classes within their code without having to dive into the actual source. It also helps clearly define exactly what each method should be doing.

File: src/com/andrew/plex/media/Media.java [hunk 3]
```diff
@@ -2,19 +2,41 @@
 
 public interface Media {
 	
-	// Plays the media
+	/**
```
Comment (thomashw) [inline src/com/andrew/plex/media/Media.java]: Comments are looking good!

So javadoc is generally used for documenting classes/methods. Mostly for people who are using your code so they don't need to dive into each class or method to understand what it's doing or expects, etc.

Regular comments should still be used within a method if you're doing something that isn't incredibly obvious.

File: src/com/andrew/plex/main/Plex.java [hunk 4]
```diff
@@ -2,33 +2,34 @@
...
+			String input = scanner.next();	
+			
 			switch (input) {
 			
-			case "1":	for( TV episode : tv.getAllShowsArray() )
-							episode.play();
+			case "1":	boolean case1 = true;
+						while(case1) {
+							
+							cli.watchTVAll(tv);
```
Comment (thomashw) [inline src/com/andrew/plex/main/Plex.java]: Reading these four lines it isn't too obvious what's going on here. It seems like you're calling a method that will watch all of the TV shows, then asking what show they'd like to watch, and then watching a specific show.

It looks like `watchTVAll` prints all of the shows? In that case you could rename it to `printTVShows` or something similar to make it more obvious.

File: src/com/andrew/plex/cli/Cli.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+import com.andrew.plex.media.AllTV;
+
+public class Cli {
+	
+	public void mainMenu() {
+		
+		System.out.println("Welcome to Plex!");
+		System.out.println();
```
Comment (thomashw) [inline src/com/andrew/plex/cli/Cli.java]: If you wanted to you could put `\n` at the end of your previous println string so this isn't needed. No big deal either way.

File: src/com/andrew/plex/cli/Cli.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+		System.out.println( "--Num-- --Name--       --Length--   --Resolution-- --File--");
+		
+		for( int i = 0 ; i < tv.getSize() ; i++ ) {
+			
+			String nameToLength = new String();
+			String name = new String();
+			
+			name = tv.getShow(i).getName();
+			
+			for( int j = 0 ; j < (15 - name.length()) ; j++)
```
Comment (thomashw) [inline src/com/andrew/plex/cli/Cli.java]: Comment explaining your `15 - name.length()` would be useful.

File: src/com/andrew/plex/cli/Cli.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+								name + nameToLength +
+								lengthHumanReadable(tv.getShow(i).getLength()) + "   " +
+								tv.getShow(i).getResolution() + "       " +
+								tv.getShow(i).getContainer() );
+		}	
+	}
+	
+	public Boolean watchTVShow(AllTV tv, int showNum) {
+		
+		if( 0 <= showNum && showNum < tv.getSize() ) {
```
Comment (thomashw) [inline src/com/andrew/plex/cli/Cli.java]: Comment.

File: src/com/andrew/plex/cli/Cli.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+		}
+	}
+	
+	private String lengthHumanReadable(int l) {
+
+		int hours = 0; 
+		int minutes = 0; 
+		int seconds = 0;
+		
+		while( l >= 3600 ) {
```
Comment (thomashw) [inline src/com/andrew/plex/cli/Cli.java]: Comments in here.

File: src/com/andrew/plex/cli/Cli.java (hunk 5 above)
Comment (andrewwinkler) [reply src/com/andrew/plex/cli/Cli.java]: I thought about that but I wanted the code to actually look like the CLI would. 
Not sure which way is better.

File: src/com/andrew/plex/main/Plex.java (hunk 4 above)
Comment (andrewwinkler) [reply src/com/andrew/plex/main/Plex.java]: I completely agree. Updated to printTVAll.

File: src/com/andrew/plex/cli/Cli.java (hunk 1 above)
Comment (andrewwinkler) [reply src/com/andrew/plex/cli/Cli.java]: Updated

File: src/com/andrew/plex/cli/Cli.java (hunk 6 above)
Comment (andrewwinkler) [reply src/com/andrew/plex/cli/Cli.java]: Agreed, updated.

File: src/com/andrew/plex/cli/Cli.java (hunk 2 above)
Comment (andrewwinkler) [reply src/com/andrew/plex/cli/Cli.java]: Very good call.
Have learned about Javadoc.

File: src/com/andrew/plex/cli/Cli.java (hunk 7 above)
Comment (andrewwinkler) [reply src/com/andrew/plex/cli/Cli.java]: Commented.

File: src/com/andrew/plex/cli/Cli.java (hunk 8 above)
Comment (andrewwinkler) [reply src/com/andrew/plex/cli/Cli.java]: Commented.

File: src/com/andrew/plex/media/Media.java (hunk 3 above)
Comment (andrewwinkler) [reply src/com/andrew/plex/media/Media.java]: That makes sense. I have so much blue and green text it's a bonanza in here.

File: src/com/andrew/plex/cli/Cli.java [hunk 9]
```diff
@@ -14,21 +22,30 @@ public void mainMenu() {
...
+		// Loop through the array list of shows.
 		for( int i = 0 ; i < tv.getSize() ; i++ ) {
 			
-			String nameToLength = new String();
 			String name = new String();
-			
 			name = tv.getShow(i).getName();
 			
+			// Add spaces between the "--Name--" and "--Length--" headings. The spaces are stored in nameToLength.
+			// 15 character is the total distance between "--Name--" and "--Length--" so the total spaces will be 15 minus the show name length.
```
Comment (thomashw) [inline src/com/andrew/plex/cli/Cli.java]: General rule is to keep row width to less than 80 characters. That's just so people can read everything without having to scroll horizontally. Most good editors can show a visual bar at the 80th character so you can see where that is. A lot of editors can also format your code for you, which will include wrapping any text that goes beyond the 80 character limit.

See here: https://stackoverflow.com/questions/3697287/eclipse-set-maximum-line-length-for-auto-formatting/3697302#3697302

and here: https://stackoverflow.com/questions/1248895/is-there-an-eclipse-line-width-marker

Comment (andrewwinkler) [reply src/com/andrew/plex/cli/Cli.java]: That makes sense. 80 character limit has been reflected in the code.

**Pull request merged**